substrate/frame/system/src/extensions/check_nonce.rs

```diff
@@ -50,11 +53,58 @@ use sp_weights::Weight;
 #[scale_info(skip_type_params(T))]
 pub struct CheckNonce<T: Config>(#[codec(compact)] pub T::Nonce);
 
+/// For a valid transaction the provides and requires information related to the nonce.
+pub struct ValidNonceInfo {
+	/// The encoded `provides` used for this transaction.
+	pub provides: Vec<Vec<u8>>,
+	/// The encoded `requires` used for this transaction.
+	pub requires: Vec<Vec<u8>>,
+}
+
 impl<T: Config> CheckNonce<T> {
 	/// utility constructor. Used only in client/factory code.
 	pub fn from(nonce: T::Nonce) -> Self {
 		Self(nonce)
 	}
+
+	/// In transaction extension, validate nonce for account, on success returns provides and
+	/// requires.
+	pub fn validate_nonce_for_account(
+		who: &T::AccountId,
+		nonce: T::Nonce,
+	) -> Result<ValidNonceInfo, TransactionValidityError> {
+		let account = crate::Account::<T>::get(who);
+		if account.providers.is_zero() && account.sufficients.is_zero() {
+			// Nonce storage not paid for
+			return Err(InvalidTransaction::Payment.into())
+		}
+		if nonce < account.nonce {
+			return Err(InvalidTransaction::Stale.into())
+		}
+
+		let provides = vec![Encode::encode(&(who.clone(), nonce))];
+		let requires = if account.nonce < nonce {
+			vec![Encode::encode(&(who.clone(), nonce.saturating_sub(One::one())))]
+		} else {
```

Comment:
It doesn't look like this condition will ever be met given the check in line 81.

Reply:
It can happen if they're equal.